keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,404 @@
+
+### Version Skew Strategy
+
+kube-controller-manager, kube-scheduler and kube-apiserver will need enable the feature gate for the full featureset
+to be present. 
+
+If kube-apiserver doesn't enable the feature gate, this new affinity will be ignored.
+
+If kube-apiserver enable the feature gate, and:
+if only the kube-controller-manager enables the feature gate, the node affinity will be ignored during the scheduling phase.
+if only the kube-scheduler enables the feature gate, the node affinity will be ignored during the execution phase. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeAffinityController 
+  - Components depending on the feature gate: kube-controller-manager, kube-scheduler, kube-apiserver
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The feature can be disabled in Alpha and Beta versions by restarting kube-controller-manager, kube-scheduler and kube-apiserver with the feature-gate off. 
+In terms of Stable versions, users can choose to opt-out by not setting the corresponding field.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The `RequiredDuringSchedulingRequiredDuringExecution` node affinity will take effect as expected.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The change is opt-in, it doesn't impact already running workloads. 
+
+###### What specific metrics should inform a rollback?
+
+Each time the NodeAffinityController evicts a pod, it will add an event with reason "NodeLabelsManagerEviction" to the cluster's events. When user see lots of this type of events, they should pay attention to this feature.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Yes.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Inspect the nodeAffinity.RequiredDuringSchedulingRequiredDuringExecution configuration
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: NodeLabelsManagerEviction
+- [ ] API .status
+  - Condition name: 
```

Comment:
This would be useful. Should it be a DisruptionTarget? https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions

Probably, since it's so similar to the Taint eviction. Any concerns @mimowo?

Reply:
+1.